### DISCUSSION TITLE
Correct metric type for ratelimit-remaining returned from an API.

### DISCUSSION THREAD

**Discussion by moolitayer** [open-telemetry/opentelemetry-go#2965] [Q&A] [answered]
I'd like to report a new metric measuring `ratelimit-remaining` headers returning from external APIs with tags for things like what is the API and who are the tenants. These return from APIs on different processes and Machines. The values go up and down over time. It looks like my only way of reporting this is using a gaugeObserver.  To do that I would have to save a mapping between tag set and the associated values and return those values from my ObserverFunc which seems convoluted. Is there a way of reporting a gauge without a callback?

What am I missing here?

**Comment by hanyuancheung** [comment, marked as answer]
The metrics part is not stable yet, you could use Prometheus to report during this time.
